docs/painless/painless-contexts/painless-watcher-condition-context.asciidoc

```diff
@@ -1,38 +1,146 @@
 [[painless-watcher-condition-context]]
 === Watcher condition context
 
+include::painless-watcher-context-variables.asciidoc[]
+
+==== Watch condition
+
 Use a Painless script as a {xpack-ref}/condition-script.html[watcher condition]
-to test if a response is necessary.
+to test if a watch or a particular action should be executed. The condition
+will return a `boolean` value to determine if the condition was met. A
+condition can be used to short circuit an entire watch or within an action
+to determine execution of said action.
 
-*Variables*
+*Return*
 
-`params` (`Map`, read-only)::
-        User-defined parameters passed in as part of the query.
+`boolean`::
+        Expects `true` if the condition is met, and `false` otherwise.
 
-`ctx['watch_id']` (`String`, read-only)::
-        The id of the watch.
+*API*
 
-`ctx['execution_time']` (`ZonedDateTime`, read-only)::
-        The start time for the watch.
+The standard <<painless-api-reference, Painless API>> is available.
 
-`ctx['trigger']['scheduled_time']` (`ZonedDateTime`, read-only)::
-        The scheduled trigger time for the watch.
+The first example is a condition for the watch itself. This determines if any
+of the actions in the watch are executed. The example aggregates the total sold
+seats for each play in the data set. The condition validates that there is at
+least one play that has sold under $15,000 or over $50,000.
 
-`ctx['trigger']['triggered_time']` (`ZonedDateTime`, read-only)::
-        The actual trigger time for the watch.
+[source,Painless]
+----
+POST _watcher/watch/_execute
+{
+  "watch" : {
+    "trigger" : { "schedule" : { "interval" : "24h" } },
+    "input" : {
+      "search" : {
+        "request" : {
+          "indices" : [ "seats" ],
+          "body" : {
+            "query" : {
+              "term": { "sold": "true"}
+            },
+            "aggs" : {
+              "theatres" : {
+                "terms" : { "field" : "play" },
+                "aggs" : {
+                  "money" : {
+                    "sum": { "field" : "cost" }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "condition" : {
+      "script" :
+      """
+        return ctx.payload.aggregations.theatres.buckets.stream()       <1>
+          .filter(t -> t.money.value < 15000 || t.money.value > 50000)  <2>
+          .collect(Collectors.toList())                                 <3>
+          .size() > 0                                                   <4>
+      """
+    },
+    "actions" : {
+      "my_log" : {
+        "logging" : {
+          "text" : "The output of the search was : {{ctx.payload.aggregations.theatres.buckets}}"
+        }
+      }
+    }
+  }
+}
+----
 
-`ctx['metadata']` (`Map`, read-only)::
-        Any metadata associated with the watch.
+<1> The Java Stream API is used in the condition. This API allows manipulation of
+the elements of the list in a pipeline.
```

Comment:
Again, avoid passive when possible. I'd say how or why the API is used, rather than just that it is. Maybe:

<1> The Java Stream API enables you to manipulate the elements of the list in a pipeline.